pkg/plugins/policies/xds/endpoints.go

```diff
@@ -0,0 +1,53 @@
+package xds
+
+import (
+	clusterv3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	endpointv3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	envoy_resource "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+
+	"github.com/kumahq/kuma/pkg/core/xds"
+	"github.com/kumahq/kuma/pkg/xds/generator"
+	"github.com/kumahq/kuma/pkg/xds/generator/egress"
+)
+
+type EndpointMap map[xds.ServiceName][]*endpointv3.ClusterLoadAssignment
+
+func GatherEndpoints(rs *xds.ResourceSet) EndpointMap {
```

Comment:
Just a random thought unrelated to this. Seems like we do these `Gather` quite often which probably means that `xds.ResourceSet` is not in a straightforward layout. Should we open an issue to rework xds.ResourceSet?

Reply:
Tracked in: https://github.com/kumahq/kuma/issues/6417